### PR TITLE
CIV-7829 Payment Succesful notifications

### PR DESCRIPTION
**Pull request by miguelMolina3691**
### Change description ###
image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/common/models/claim.ts [hunk 1]
```diff
@@ -138,6 +139,7 @@ export class Claim {
   specRespondent1Represented?: YesNoUpperCamelCase;
   respondentPaymentDeadline: Date;
   respondentSignSettlementAgreement?: GenericYesNo;
+  hearingFeePaymentDetails: PaymentDetails;
```
Comment (laarmada) [inline src/main/common/models/claim.ts:142]: it should be under caseProgression ---> xxx---..... no in tue root of claim

File: src/main/routes/urls.ts [hunk 2]
```diff
@@ -266,4 +266,8 @@ export const HEARING_FEE_APPLY_HELP_FEE_HEARING_FEE = `${BASE_CASE_PROGRESSION_U
 export const APPLY_HELP_WITH_FEES_START = `${APPLY_HELP_WITH_FEES}/start`;
 export const APPLY_HELP_WITH_FEES_REFERENCE = `${APPLY_HELP_WITH_FEES}/reference-number`;
 export const PAY_HEARING_FEE_UNSUCCESSFUL_URL = `${BASE_CASE_PROGRESSION_URL}/payment-unsuccessful`;
-export const HEARING_FEE_CONFIRMATION_URL = `${PAY_HEARING_FEE_URL}/confirmation`;
\ No newline at end of file
+export const HEARING_FEE_CONFIRMATION_URL = `${PAY_HEARING_FEE_URL}/confirmation`;
+export const HEARING_FEE_MAKE_PAYMENT_AGAIN = `${BASE_CASE_PROGRESSION_URL}/make-payment-again`;
```
Comment (laarmada) [inline src/main/routes/urls.ts:270]: add _URL at the end

File: src/main/services/dashboard/getDashboardContent.ts [hunk 3]
```diff
@@ -24,8 +25,18 @@ export const getClaimantNotifications = (claim: Claim, lng: string) => {
       .build())
     .build();
 
+  const hearingFeePaidNotification = new NotificationBuilder()
+    .addTitle(t('PAGES.LATEST_UPDATE_CONTENT.CASE_PROGRESSION.HEARING_FEE.PAID_NOTIFICATION_TITLE', { lng }))
```
Comment (laarmada) [inline src/main/services/dashboard/getDashboardContent.ts:29]: I think we dont need {} could you check it?

File: src/main/services/features/caseProgression/hearingFee/applyHelpFeeSelectionService.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import {constructResponseUrlWithIdParams} from 'common/utils/urlFormatter';
+import {getFeePaymentRedirectInformation} from 'services/features/feePayment/feePaymentService';
+import {FeeType} from 'form/models/helpWithFees/feeType';
+
+const paymentInformation = 'paymentInformation';
+const hearing = 'hearing';
+const hearingFeeHelpSelection = 'hearingFeeHelpSelection';
+
+export const getRedirectUrl = async (claimId: string, IsApplyHelpFeeModel: GenericYesNo, req: AppRequest): Promise<string> => {
+  const redisClaimId = generateRedisKey(<AppRequest>req);
```
Comment (laarmada) [inline src/main/services/features/caseProgression/hearingFee/applyHelpFeeSelectionService.ts:16]: where is the try/catch function?

File: src/main/services/features/caseProgression/hearingFee/makePaymentAgainService.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import {generateRedisKey} from 'modules/draft-store/draftStoreService';
+import {getFeePaymentRedirectInformation} from 'services/features/feePayment/feePaymentService';
+import {FeeType} from 'form/models/helpWithFees/feeType';
+
+import {saveCaseProgression} from 'services/features/caseProgression/caseProgressionService';
+
+const paymentInformation = 'paymentInformation';
+const hearing = 'hearing';
+export const getRedirectUrl = async (claimId: string,  req: AppRequest): Promise<string> => {
+  const redisClaimId = generateRedisKey(<AppRequest>req);
```
Comment (laarmada) [inline src/main/services/features/caseProgression/hearingFee/makePaymentAgainService.ts:12]: try/catch also missing

File: src/main/services/features/caseProgression/hearingFee/paymentConfirmationService.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+const paymentInformation = 'paymentInformation';
+const hearing = 'hearing';
+const success = 'Success';
+const failed = 'Failed';
+const paymentCancelledByUser = 'Payment was cancelled by the user';
+
+export const getRedirectUrl = async (claimId: string, req: AppRequest): Promise<string> => {
+
+  const redisClaimId = generateRedisKey(<AppRequest>req);
```
Comment (laarmada) [inline src/main/services/features/caseProgression/hearingFee/paymentConfirmationService.ts:21]: try/catch function to manage and exception control

**Review by laarmada — CHANGES_REQUESTED** [review]
please talk with kannan about his changes if exist @kannan-v-hmcts

File: src/main/services/features/caseProgression/hearingFee/applyHelpFeeSelectionService.ts (hunk 4 above)
Comment (kannan-v-hmcts) [reply src/main/services/features/caseProgression/hearingFee/applyHelpFeeSelectionService.ts:16]: Hi @laarmada This function is called within a try block from the controller and the exception is caught there. Do we still need try catch in this service class ?

File: src/main/routes/urls.ts (hunk 2 above)
Comment (kannan-v-hmcts) [reply src/main/routes/urls.ts:270]: This code is part of my PR - https://github.com/hmcts/civil-citizen-ui/pull/2709. I will make the changes there

File: src/main/services/features/caseProgression/hearingFee/applyHelpFeeSelectionService.ts (hunk 4 above)
Comment (laarmada) [reply src/main/services/features/caseProgression/hearingFee/applyHelpFeeSelectionService.ts:16]: once we have a await/ database/ redis/ etc external service is mandatory to create a try cath block in order to follow the exception thread and also add specific log message..... that is the correct way to do, if you have controller ---> service ---> model (call external function) we need to create in the 3 parts try /catch block in order to have an exception control .... else we dont know exactly where is the causing of root cause

Comment (kannan-v-hmcts) [reply src/main/services/features/caseProgression/hearingFee/applyHelpFeeSelectionService.ts:16]: Thanks for the explanation @laarmada. I will address this in my PR https://github.com/hmcts/civil-citizen-ui/pull/2709

File: src/main/services/features/caseProgression/hearingFee/paymentConfirmationService.ts (hunk 6 above)
Comment (kannan-v-hmcts) [reply src/main/services/features/caseProgression/hearingFee/paymentConfirmationService.ts:21]: Will be addressed in https://github.com/hmcts/civil-citizen-ui/pull/2709

File: src/main/services/features/caseProgression/hearingFee/makePaymentAgainService.ts (hunk 5 above)
Comment (kannan-v-hmcts) [reply src/main/services/features/caseProgression/hearingFee/makePaymentAgainService.ts:12]: Will be addressed in https://github.com/hmcts/civil-citizen-ui/pull/2709

**Comment by kannan-v-hmcts** [comment]
@laarmada @miguelMolina3691  Changes suggested by Sergio are done and pushed to my PR - https://github.com/hmcts/civil-citizen-ui/pull/2709/commits/578f024a1f6286af86eacb607e0aa4ef2182ee80

**Review by dharmendrak — APPROVED** [review]

**Review by m-meulendijks-v1 — COMMENTED** [review]
I am noticing a lot of references to alternative contact person + Carm. Should those changes be in here if they are not already in feature/Dashboard_dev?

File: src/main/services/dashboard/getDashboardContent.ts [hunk 7]
```diff
@@ -24,8 +27,18 @@ export const getClaimantNotifications = (claim: Claim, lng: string) => {
...
+  const hearingFeePaidNotification = new NotificationBuilder()
+    .addTitle(t('PAGES.LATEST_UPDATE_CONTENT.CASE_PROGRESSION.HEARING_FEE.PAID_NOTIFICATION_TITLE', lng))
+    .addContent(new PageSectionBuilder()
+      .addParagraph(t('PAGES.LATEST_UPDATE_CONTENT.CASE_PROGRESSION.HEARING_FEE.PAID_NOTIFICATION_SUBTITLE', lng))
+      .build())
+    .build();
+
   dashboardNotificationsList.push(waitForDefendantResponseNotification);
   dashboardNotificationsList.push(waitForDefendantResponseNotification2);
+  if (claim.caseProgression?.hearing?.paymentInformation?.status === success || claim.caseProgressionHearing?.hearingFeePaymentDetails?.status === PaymentStatus.SUCCESS) {
```
[the remark this answers is not in the document]
Comment (kannan-v-hmcts) [reply src/main/services/dashboard/getDashboardContent.ts:39]: As discussed, the status is saved differently in Redis and CCD. Ignoring this comment as agreed

File: src/main/common/models/claim.ts [hunk 8]
```diff
@@ -134,9 +134,9 @@ export class Claim {
   defendantSignedSettlementAgreement?: YesNo;
   courtDecision: RepaymentDecisionType;
   feeTypeHelpRequested: FeeType;
+  helpWithFeesRequested: string;
```
[the remark this answers is not in the document]
Comment (kannan-v-hmcts) [reply src/main/common/models/claim.ts]: Done. Moved it under caseProgression -> Hearing

**Review by laarmada — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**